### PR TITLE
[uvw] Bump to 3.2.0

### PR DESCRIPTION
**Pull request by xiaozhuai**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

This pr depends on #32525

### CODE REVIEW

File: ports/uvw/vcpkg.json [hunk 1]
```diff
@@ -1,7 +1,6 @@
 {
   "name": "uvw",
-  "version": "3.0.0",
-  "port-version": 1,
+  "version": "3.2.0",
   "description": "A compilable static library, event based, tiny and easy to use libuv wrapper in modern C++.",
```
Comment (dg0yt) [inline ports/uvw/vcpkg.json:4]: This seems to need an update if it is no longer a static library.

**Comment by jimwang118** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

File: ports/uvw/vcpkg.json (hunk 1 above)
Comment (xiaozhuai) [reply ports/uvw/vcpkg.json]: Done

**Comment by jimwang118** [comment]
The following error occurs when testing usage.
```
Error		CMake Error at F:/uvw/installed/x64-windows/share/uvw/uvwConfig.cmake:60 (set_target_properties):
  The link interface of target "uvw::uvw" contains:

    libuv::uv

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.		F:/uvw/installed/x64-windows/share/uvw/uvwConfig.cmake	60
```

**Comment by xiaozhuai** [comment]
> The following error occurs when testing usage.

I'll take care of this later

**Comment by jimwang118** [comment]
Usage test pass with following triplets:

```
x86-windows
x64-windows
x64-windows-static
```

File: ports/uvw/fix-find-libuv.patch [hunk 2]
```diff
@@ -1,18 +1,16 @@
...
-+find_package(libuv CONFIG REQUIRED)
-+if (TARGET libuv::uv)
-+    target_link_libraries(uvw PRIVATE libuv::uv)
-+else()
-+    target_link_libraries(uvw PRIVATE libuv::uv_a)
-+endif()
++find_path(LIBUV_INCLUDE_DIR NAMES uv.h PATH_SUFFIXES uv REQUIRED)
++find_library(LIBUV_LIBRARY NAMES uv uv_a libuv REQUIRED)
++target_include_directories(uvw PUBLIC ${LIBUV_INCLUDE_DIR})
++target_link_libraries(uvw PRIVATE ${LIBUV_LIBRARY})
```
Comment (dg0yt) [inline ports/uvw/fix-find-libuv.patch:13]: Why is this changed? The new code is will not support multi-config generators.
What may have been missing is a `find_dependency` in the exported config.

Comment (xiaozhuai) [reply ports/uvw/fix-find-libuv.patch:13]: > What may have been missing is a `find_dependency`

You are right, I didn't find a way how to add this in exported config.
Can you give me some example?

Comment (dg0yt) [reply ports/uvw/fix-find-libuv.patch:13]: vcpkg ports are full of examples, either as part of the patch, or via script code in `portfile.cmake`.
E.g. ports/proj/fix-proj4-targets-cmake.patch or ports/libssh/portfile.cmake.

Comment (xiaozhuai) [reply ports/uvw/fix-find-libuv.patch]: @dg0yt Thanks!

**Comment by BillyONeal** [comment]
Thanks for the update!
